### PR TITLE
feat: add end-user messages to HTTP errors

### PR DESCRIPTION
**Pull request by jvallesm**
Because

- Connector errors in the pipeline are often unclear to end users and expose implementation details.
- Each connector has similar functionality implemented by duplicating and adapting code.

This commit

- Extracts the common logic to call connector vendor APIs.
  - `httpclient` is a client that provides default configuration configuration, logging and error wrapping.
- Migrates API connectors (i.e. connectors whose triggers make an API call) to use the `httpclient` package.
  - [x] OpenAI
  - [x] Pinecone
  - [x] Stable diffussion
  - [x] Hugging face
  - [x] Instill AI
  - [x] REST API 
- Basic coverage is added along the way.

## 🗒️ Notes

[resty](https://github.com/go-resty/resty) was chosen for several reasons:
- Great API to allow clients to define the request and response format with `SetBody`, `SetResult` and `SetError`.
- Hooks that allow us to focus on the business logic: log and wrap around errors.
- Automatic content type detection.
- Client configuration that can be overriden by request.

Overall it produces more compact and readable code than building methods around `http.Client.Do`.

One drawback: the [`OnError` hook ](https://github.com/go-resty/resty/blob/v2/client.go#L1194)doesn't allow us to modify the error that's returned by `client.R().Execute()`. This means that [if `http.Client.Do` returns an error](https://github.com/go-resty/resty/blob/v2/client.go#L1194), we can't attach an end-user error. For non-2xx responses, `OnAfterRequest` is executed and the error is successfully wrapped.

On Instill AI connector, most interactions take place via gRPC. We'll need to address gRPC client errors in a similar PR, here it only affects to the `Test` action. An HTTP client that was a property of the Executor was removed as unused.

## 🔨 QA

Each task was executed both successfully and causing an API failure (e.g. wrong parameter).

### CODE REVIEW

File: pkg/restapi/auth.go [hunk 1]
```diff
@@ -1,123 +1,98 @@
...
+type authType string
 
 const (
-	NoAuthType      AuthType = "NO_AUTH"
-	BasicAuthType   AuthType = "BASIC_AUTH"
-	APIKeyType      AuthType = "API_KEY"
-	BearerTokenType AuthType = "BEARER_TOKEN"
+	noAuthType      authType = "NO_AUTH"
+	basicAuthType   authType = "BASIC_AUTH"
+	apiKeyType      authType = "API_KEY"
```
[the remark this answers is not in the document]
Comment (jvallesm) [reply pkg/restapi/auth.go:15]: types, not credentials

File: pkg/instill/main.go [hunk 2]
```diff
@@ -202,10 +139,43 @@
...
+}
+
+func newHTTPClient(config *structpb.Struct, logger *zap.Logger) *httpclient.Client {
+	c := httpclient.New("Instill AI", getServerURL(config),
+		httpclient.WithLogger(logger),
+		httpclient.WithEndUserError(new(errBody)),
+	)
+
+	c.SetTransport(&http.Transport{
+		TLSClientConfig:   &tls.Config{InsecureSkipVerify: true},
```
[the remark this answers is not in the document]
Comment (jvallesm) [reply pkg/instill/main.go:168]: cc @donch1989 I just replicated the existing transport configuration. Can we remove this or is there any business logic behind it?

[the remark this answers is not in the document]
Comment (donch1989) [reply pkg/instill/main.go:168]: I remembered we will encounter some issue without this. We can keep it for now and solve it later.

[the remark this answers is not in the document]
Comment (donch1989) [reply pkg/instill/main.go:168]: Created a [issue](https://github.com/instill-ai/community/issues/519) for this

**Comment by donch1989** [comment]
LGTM

**Pull request merged**